src/MissionParallelCatchup/parallel_catchup_helm/templates/catchup_workers.yaml

```diff
@@ -32,12 +32,32 @@ spec:
           requiredDuringSchedulingIgnoredDuringExecution:
             nodeSelectorTerms:
             - matchExpressions:
+              {{- range .Values.worker.requireNodeLabels }}
+              - key: {{ .key }}
+                operator: {{ .operator }}
+                {{- with .values }}
+                values: {{ toJson . }}
+                {{- end }}
+              {{- end }}
+              {{- range .Values.worker.avoidNodeLabels }}
+              - key: {{ .key }}
+                operator: {{ .operator }}
+                {{- with .values }}
+                values: {{ toJson . }}
+                {{- end }}
+              {{- end }}
+              {{- if not .Values.worker.unevenSched }}
               - key: node-role.kubernetes.io/control-plane
                 operator: DoesNotExist
-              - key: purpose
-                operator: NotIn
-                values:
-                - ssc
+              {{- end }}
+      tolerations:
+      {{- range .Values.worker.tolerateNodeTaints }}
+      - key: {{ .key }}
+        operator: {{ .operator }}
+        {{- with .value }}
+        value: "{{ . }}"
+        {{- end }}
+      {{- end }}
```

Comment:
I tested this part and we end up with slightly incorrect syntax here. The `operator` and `value` are not needed for tolerations but we need `effect`. Her is an example of what we need:
```
      tolerations:
        - effect: NoSchedule
          key: mykey
```